src/scripts/main.js

```diff
@@ -1,3 +1,44 @@
 'use strict';
 
-// write code here
+const ul = document.querySelector('.tree');
+
+// We create a span in li that contains a list
+function addSpan(elem) {
+  const elementChildren = [...elem.children];
+
+  if (elementChildren.length === 0) {
+    return;
+  }
+
+  if (elementChildren.length === 1) {
+    const span = document.createElement('span');
+    const textLi = elem.childNodes[0].textContent.trim();
+
+    span.textContent = textLi;
+    elem.prepend(span);
+    elem.childNodes[1].remove();
+  }
+
+  for (const el of elementChildren) {
+    addSpan(el);
+  }
+}
+
+addSpan(ul);
+
+// We hide ul when clicking on span
+document.addEventListener('click', (e) => {
+  const target = e.target.closest('span');
+
+  if (!target) {
+    return;
+  }
+
+  const list = target.nextSibling;
+
+  if (list.hasAttribute('hidden')) {
```

Comment:
The 'list' variable might not always be an element with the 'hasAttribute' method if 'target.nextSibling' is not the expected element. Consider adding a check to ensure 'list' is an element node before calling 'hasAttribute'.